aten/src/TH/generic/THTensorMoreMath.cpp

```diff
@@ -28,9 +28,21 @@ int THTensor_(equal)(THTensor *ta, THTensor* tb)
     scalar_t *tbp = tb->data<scalar_t>();
     ptrdiff_t sz = THTensor_(nElement)(ta);
     ptrdiff_t i;
-    for (i=0; i<sz; ++i){
-      if(tap[i] != tbp[i]) return 0;
-    }
+    at::parallel_for(
+        0,
+        sz,
+        TH_OMP_OVERHEAD_THRESHOLD,
+        [&](int64_t begin, int64_t end) {
+          for (auto iter = begin; iter < end; iter++) {
+            if (!equal) {
```

Comment:
I'm a bit concerned about read/writes, in some cases they will be atomic but I'm not sure this is always true;
could you use local to the scope equal variable and then write it into atomic int variable defined outside

Reply:
In case of non atomic write worst thing that could happen is making few more loop iterations. It is much cheaper than any synchronization.

Reply:
what about read/write race?

Reply:
using std::atomic<int> would probably be also negligibly different as it would just use atomic processor instructions

Reply:
Updated to `std::atomic<int>`.